crates/rpc/rpc-eth-api/src/helpers/pending_block.rs

```diff
@@ -163,10 +183,63 @@ pub trait LoadPendingBlock: EthApiTypes {
             };
 
             let now = Instant::now();
-            *lock = Some(PendingBlock::new(pending_block.clone(), now + Duration::from_secs(1)));
+            *lock = Some(PendingBlock::new(
+                sealed_block.clone(),
+                now + Duration::from_secs(1),
+                receipts.clone(),
```

Comment:
If the pending block is built with receipts now, it shouldn't be necessary to return the receipts separately, so this clone should be avoidable, no? Ref https://github.com/paradigmxyz/reth/pull/10070#discussion_r1703342665

Reply:
I do not see a way to avoid the clones tbh, unless I change the signature of the func

Reply:
ah yeah see what you mean. return `&Mutex<Option<PendingBlock>>` instead of `SealedBlockWithSenders`.

Reply:
or don't return any data at all rather, so `impl Future<Output = Result<(), Self::Error>> + Send`. then call `self.pending_block()` again to read the updated pending block.

Reply:
no mutex in function signatures, this clone is appropriate